src/cmr/sizing/formats.clj

```diff
@@ -29,20 +35,31 @@
       (get-in [:umm :SizeEstimation :AverageSizeOfGranulesSampled])
       read-number))
 
-(defn- get-avg-compression-rate-ascii
-  "Gets SizeEstimation value for AvgCompressionRateASCII and parses it to a number."
-  [variable]
-  (-> variable
-      (get-in [:umm :SizeEstimation :AvgCompressionRateASCII])
-      read-number))
-
 (defn- get-avg-compression-rate-netcdf4
   "Gets SizeEstimation value for AvgCompressionRateNetCDF4 and parses it to a number."
   [variable]
   (-> variable
       (get-in [:umm :SizeEstimation :AvgCompressionRateNetCDF4])
       read-number))
 
+(defn- get-avg-fill-value-digit-number
+  "Get the average digit number for each value of FillValues in the variable.
+  For example, FillValues containing [{:Value 1} {:Value 100}], returns (1+3)/2=2."
+  [variable]
+  (let [fvs (get-in variable [:umm :FillValues])
+        fvs-digit-numbers (map #(count (str (:Value %))) fvs)]
+    (/ (reduce + fvs-digit-numbers) (count fvs-digit-numbers))))
+
+(defn- get-avg-valid-range-digit-number
+  "Get the average digit number for each range of ValidRanges in the variable.
+  For example, ValidRanges containing [{:Min 0 :Max 100} {:Min 10 :Max 1000}] returns
+  (1+3+2+4)/4=2.5"
+  [variable]
+  (let [vrs (get-in variable [:umm :ValidRanges])
+        vrs-min-max (concat (map :Min vrs) (map :Max vrs))
+        vrs-digit-numbers (map #(count (str %)) vrs-min-max)]
```

Comment:
Similar comment about dividing by zero if ValidRanges are an optional field in the metadata.

If they are optional I think we should default to using the fill value here when not provided. If both fill value and valid ranges are not provided I'd probably default to 3 again.

Reply:
Done.